addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1361,3 +1370,80 @@ func (r *ClusterBootstrapReconciler) reconcileClusterProxyAndNetworkSettings(clu
 
 	return nil
 }
+
+func (r *ClusterBootstrapReconciler) reconcileDelete(cluster *clusterapiv1beta1.Cluster, log logr.Logger) (ctrl.Result, error) {
+
+	remoteClient, err := util.GetClusterClient(r.context, r.Client, r.Scheme, clusterapiutil.ObjectKey(cluster))
+	clusterBootstrap := &runtanzuv1alpha3.ClusterBootstrap{}
+	err = remoteClient.Get(r.context, client.ObjectKeyFromObject(cluster), clusterBootstrap)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if clusterBootstrap == nil {
+		return ctrl.Result{}, nil
+	}
+
+	err = r.removeAdditionalPackageInstalls(cluster, log)
```

Comment:
If the remote cluster is being deleted, meaning all the resources on it will be gone. I assume we probably don't need to clean up the resources in the remote cluster. We just need to clean up all the resources we created in the mgmt cluster.

Reply:
Question resolved: Cluster deletion should trigger delete of all additionalPackages installed in a cluster so that external resources such as LoadBalancers and PersistentVolumes are deleted. Should not delete Kapp, CNI, CPI and CSI packages on the target cluster.